File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/dagger/module/RealmModule.java
package com.lattis.ellipse.presentation.dagger.module;

import android.content.Context;

import com.lattis.ellipse.data.database.DatabaseManager;
import com.lattis.ellipse.domain.repository.DataBaseManager;
import com.lattis.ellipse.presentation.dagger.qualifier.DatabaseName;
import com.lattis.ellipse.presentation.dagger.qualifier.DatabaseSchemaVersion;
import com.lattis.ellipse.presentation.dagger.qualifier.FleetId;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.realm.Realm;
import io.realm.RealmConfiguration;

@Module
public class RealmModule {

    private static final String DATABASE_NAME = "ellipse.db";
    private static final String FLEET_ID = "9";
    private static final int DATABASE_SCHEMA_VERSION = 1;

    @Provides
    @Singleton
    @DatabaseName
    String provideDatabaseName() {
        return DATABASE_NAME;
    }

    @Provides
    @Singleton
    @DatabaseSchemaVersion
    int provideDatabaseSchemaVersion() {
        return DATABASE_SCHEMA_VERSION;
    }

    @Provides
    @Singleton
    RealmConfiguration provideRealmConfiguration() {
        Realm realm = Realm.getDefaultInstance();
        RealmConfiguration configuration = realm.getConfiguration();
        realm.close();
        return configuration;
    }

    @Provides
    @Singleton
    @FleetId
    String provideFleetId(){
        return FLEET_ID;
    }

    @Provides
    @Singleton
    DataBaseManager provideDatabaseManager(Context context, @DatabaseName String databaseName, @DatabaseSchemaVersion int schemaVersion){
        return new DatabaseManager(context, databaseName, schemaVersion);
    }


}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/infrastructure/di/module/RealmModule.kt
package com.lattis.lattis.infrastructure.di.module

import android.content.Context
import com.lattis.data.repository.implementation.api.DataBaseRepositoryImp
import com.lattis.domain.repository.DataBaseRepository
import dagger.Module
import dagger.Provides
import io.realm.Realm
import io.realm.RealmConfiguration
import javax.inject.Named
import javax.inject.Singleton

@Module
class RealmModule {
    @Provides
    @Singleton
    @Named("DatabaseName")
    fun provideDatabaseName(): String {
        return DATABASE_NAME
    }

    @Provides
    @Singleton
    @Named("DatabaseSchemaVersion")
    fun provideDatabaseSchemaVersion(): Int {
        return DATABASE_SCHEMA_VERSION
    }

    @Provides
    @Singleton
    fun provideRealmConfiguration(): RealmConfiguration {
        val realm = Realm.getDefaultInstance()
        val configuration = realm.configuration
        realm.close()
        return configuration
    }

    @Provides
    @Singleton
    @Named("UUID")
    fun provideFleetId(): String {
        return UUID
    }

    @Provides
    @Singleton
    fun provideDataBaseRepository(context: Context, @Named("DatabaseName") databaseName: String, @Named("DatabaseSchemaVersion") schemaVersion: Int): DataBaseRepository{
        return DataBaseRepositoryImp(context!!, databaseName!!, schemaVersion)
    }

    companion object {
        private const val DATABASE_NAME = "lattis.db"
        private const val UUID = "LATTIS_APP"
        private const val DATABASE_SCHEMA_VERSION = 1
    }
}